src/AnyAddress.h

```diff
@@ -34,125 +36,96 @@
 namespace TW {
 
 class AnyAddress {
-  public:
+public:
     static auto dataFromString(const std::string& string, TWCoinType coin) -> Data {
-        switch (coin) {
-        case TWCoinTypeBinance:
-        case TWCoinTypeCosmos:
-        case TWCoinTypeKava:
-        case TWCoinTypeTerra:
-        case TWCoinTypeBandChain:
-        case TWCoinTypeTHORChain:
-        case TWCoinTypeBluzelle:
-        case TWCoinTypeIoTeX:
-        case TWCoinTypeCryptoOrg:
-        case TWCoinTypeOsmosis:
-        case TWCoinTypeHarmony:
-        case TWCoinTypeNativeEvmos:
-            {
-                Cosmos::Address addr;
-                if (!Cosmos::Address::decode(string, addr)) {
-                    break;
-                }
-                return addr.getKeyHash();
+        const auto blockchain = TW::blockchain(coin);
+        switch (blockchain) {
+        case TWBlockchainBinance:
+        case TWBlockchainCosmos:
+        case TWBlockchainThorchain:
+        case TWBlockchainIoTeX:
+        case TWBlockchainHarmony: {
+            Cosmos::Address addr;
+            if (!Cosmos::Address::decode(string, addr)) {
+                break;
             }
+            return addr.getKeyHash();
+        }
 
-        case TWCoinTypeBitcoin:
-        case TWCoinTypeDigiByte:
-        case TWCoinTypeGroestlcoin:
-        case TWCoinTypeLitecoin:
-        case TWCoinTypeViacoin:
-            {
+        case TWBlockchainBitcoin:
+        case TWBlockchainGroestlcoin:
+            switch (coin) {
+            case TWCoinTypeBitcoin:
+            case TWCoinTypeDigiByte:
+            case TWCoinTypeGroestlcoin:
+            case TWCoinTypeLitecoin:
+            case TWCoinTypeViacoin: {
                 auto decoded = Bitcoin::SegwitAddress::decode(string);
                 if (!std::get<2>(decoded)) {
                     break;
                 }
                 return std::get<0>(decoded).witnessProgram;
             }
 
-        case TWCoinTypeBitcoinCash: {
-            Data data;
-            auto addr = Bitcoin::BitcoinCashAddress(string);
-            data.resize(Bitcoin::Address::size);
-            size_t outlen = 0;
-            cash_data_to_addr(data.data(), &outlen, addr.bytes.data(), 34);
-            return {data.begin() + 1, data.end()};
-        }
+            case TWCoinTypeBitcoinCash: {
+                Data data;
+                auto addr = Bitcoin::BitcoinCashAddress(string);
+                data.resize(Bitcoin::Address::size);
```

Comment:
Do you really want to use `resize` here? `resize` appends the default element to the vector, and `reserve` grow the capacity but let the data uninitialized.

If you really want to use `resize` then you can write `Data data(Bitcoin::Address::size)` will allow to use the constructor with pre-defined size from `std::vector` 

from https://en.cppreference.com/w/cpp/container/vector/vector

Reply:
This is legacy code, good suggestion